apps/channels/forms.py

```diff
@@ -217,3 +217,20 @@ def post_save(self, channel: ExperimentChannel):
                 service.join_channel(channel_id)
             except Exception as e:
                 raise ExperimentChannelException("Failed to join the channel") from e
+
+
+class CommCareConnectChannelForm(ExtraFormBase):
+    bot_name = forms.CharField(
+        label="Bot Name",
+        help_text=(
+            "This is the name of the bot that will be displayed to users on commcare connect"
+            "Once set, this name cannot be changed."
```

Comment:
Why shouldn't it be changed? I don't think it matters if they change it here since it will only get used for new users. If we don't allow them to change it they will remove the channel and then add a new one with a different name.

@calellowitz other than displaying this to the user for consent, is the `channel_source` used for anything?

Reply:
It is used for the user visible name on the mobile, so in your list of channels.

Reply:
Yeah, connectid can't tell those channels are related in any way. Source is set once at creation, but otherwise the channels are exclusively referenced by UUID, and don't share anything on the backend data model. Even if they had the same "source," they wouldn't be connected in any way.

Reply:
Updated. https://github.com/dimagi/open-chat-studio/pull/1041/commits/ca2ac38a3876e20d36c23173a82b0d0eb6bdf31e